static/app/views/settings/organizationTeams/teamDetails.tsx

```diff
@@ -189,26 +137,73 @@ class TeamDetails extends React.Component<Props, State> {
     ];
 
     return (
-      <div>
-        <SentryDocumentTitle title={t('Team Details')} orgSlug={params.orgId} />
-        <h3>
-          <IdBadge hideAvatar team={team} avatarSize={36} />
-        </h3>
-
-        <NavTabs underlined>{navigationTabs}</NavTabs>
-
-        {React.isValidElement(children) &&
-          React.cloneElement(children, {
-            team,
-            onTeamChange: this.onTeamChange,
-          })}
-      </div>
+      <Teams slugs={[params.teamId]}>
+        {({teams, initiallyLoaded}) => (
+          <div>
+            {initiallyLoaded ? (
+              teams.length ? (
+                teams.map((team, i) => {
+                  if (!team || !team.hasAccess) {
+                    return (
+                      <Alert type="warning">
+                        {team ? (
+                          <RequestAccessWrapper>
+                            {tct('You do not have access to the [teamSlug] team.', {
+                              teamSlug: <strong>{`#${team.slug}`}</strong>,
+                            })}
+                            <Button
+                              disabled={requesting || team.isPending}
+                              size="small"
+                              onClick={() => this.handleRequestAccess(team)}
+                            >
+                              {team.isPending
+                                ? t('Request Pending')
+                                : t('Request Access')}
+                            </Button>
+                          </RequestAccessWrapper>
+                        ) : (
+                          <div>{t('You do not have access to this team.')}</div>
+                        )}
+                      </Alert>
+                    );
+                  }
+                  return (
+                    <div key={i}>
+                      <SentryDocumentTitle
+                        title={t('Team Details')}
+                        orgSlug={params.orgId}
+                      />
+                      <h3>
+                        <IdBadge hideAvatar team={team} avatarSize={36} />
+                      </h3>
+
+                      <NavTabs underlined>{navigationTabs}</NavTabs>
+
+                      {React.isValidElement(children) &&
+                        React.cloneElement(children, {
+                          team,
+                          onTeamChange: this.onTeamChange,
+                        })}
+                    </div>
+                  );
+                })
+              ) : (
+                <Alert type="warning">
+                  <div>{t('You do not have access to this team.')}</div>
+                </Alert>
+              )
+            ) : (
+              <LoadingIndicator />
+            )}
+          </div>
+        )}
+      </Teams>
     );
   }
 }
 
 // TODO(davidenwang): change to functional component and replace withTeams with useTeams
```

Comment:
@mikellykels I think this could be improved by taking care of this todo at the same time. Your solution via render has brought you 80% of the way there.